core/src/main/java/org/apache/iceberg/BaseAllMetadataTableScan.java

```diff
@@ -40,7 +40,7 @@ abstract class BaseAllMetadataTableScan extends BaseMetadataTableScan {
   protected abstract String tableType();
 
   @Override
-  public TableScan appendsBetween(long fromSnapshotId, long toSnapshotId) {
+  public TableScan appendsInRange(Long fromSnapshotId, long toSnapshotId) {
```

Comment:
In the future, we want to support reading overwrite snapshots so maybe we could change to `dataBetween(startSnaphotId, endSnasphotId)`.

Reply:
if the community agrees with PR #4580 , we will abort this PR/approach. In PR #4580 , we can add more flexible `IncrementalScan` options (not just append) to support use cases like CDC read.

Reply:
Cool, let me take a look.